src/main/java/com/rackspace/salus/telemetry/presence_monitor/services/ResourceListener.java

```diff
@@ -110,4 +112,11 @@ public void onPartitionsAssigned(Map<TopicPartition, Long> map, ConsumerSeekCall
     public void onIdleContainer(Map<TopicPartition, Long> map, ConsumerSeekCallback consumerSeekCallback) {
         // do nothing
     }
+
+    public ResourceDTO findResourceByTenantIdAndResourceId(String tenantId, String resourceId) {
+        Resource resource = resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
+            .orElse(null);
+
+        return resource == null ? null : new ResourceDTO(resource, null);
```

Comment:
I think Geoff also suggested this improvement in another PR.

```suggestion
        return resourceRepository.findByTenantIdAndResourceId(tenantId, resourceId)
            .map(resource -> new ResourceDTO(resource, null))
            .orElse(null);
```

Reply:
Yes....Will do this in all the other PRs